client/transactions/list/index.tsx

```diff
@@ -580,11 +580,18 @@ export const TransactionsList = (
 						)
 					);
 
-					wcpayTracks.recordEvent( 'wcpay_transactions_download', {
-						exported_transactions: exportedTransactions,
-						total_transactions: exportedTransactions,
-						download_type: downloadType,
-					} );
+					wcpayTracks.recordEvent(
+						props.depositId
+							? wcpayTracks.events
+									.DEPOSITS_DETAIL_DOWNLOAD_CSV_CLICK
+							: wcpayTracks.events
+									.TRANSACTIONS_DOWNLOAD_CSV_CLICK,
```

Comment:
I'm not sure we need a different event here – the action is the same, downloading a list of transactions as a CSV file. Could add a property to distinguish downloads from the transactions list vs. for a specific deposit. If we want to know how many merchants are downloading CSVs of transactions overall, we'd need to consider both events. 

Though I'm fine with having a separate event (as you've done here). If we do go with a separate event, I suggest naming it more clearly, i.e. indicating that it's downloading transactions. e.g. DEPOSITS_DETAIL_DOWNLOAD_TRANSACTIONS_CSV_CLICK

Reply:
> I'm not sure we need a different event here ... add a property to distinguish downloads from the transactions list
+1

It's a bit of an unknown, but there could be a situation in the future where this transaction list can be downloaded from other locations (e.g. overview page). If this occurs, it would be simpler to add an event prop for each new location, rather than a new event for each.

Are there any downsides to using an event prop rather than a separate event here?

Reply:
> Are there any downsides to using an event prop rather than a separate event here?

My main reasoning was that it's unlikely that someone wants to know when the transaction list CSV is downloaded. 
A product owner looking at these stats will be looking at them to make product decisions. Are these flows working? Are these features being used?
Downloading all transactions and downloading transactions that make up a deposit are two separate flows/features. 

>  I suggest naming it more clearly, i.e. indicating that it's downloading transactions. e.g. DEPOSITS_DETAIL_DOWNLOAD_TRANSACTIONS_CSV_CLICK

After considering this I opted to keep with the naming convention in line with other CSV download events.

Reply:
> After considering this I opted to keep with the naming convention in line with other CSV download events.

Assuming you mean the name is `DEPOSITS_DETAIL_DOWNLOAD_CSV_CLICK `? I'd argue `DEPOSITS_DETAIL_TRANSACTIONS_DOWNLOAD_CSV_CLICK` or similar is _more_ consistent, since both buttons download a list of transactions; "deposits detail" is not getting downloaded, that's the event source. Not a big deal though.

Reply:
This might be minor, but I noticed that the placement of the call to `recordEvent` means that it may not be recorded if there is an error in downloading.

This can be confirmed by watching the network requests while clicking: the `t.gif` request doesn't doesn't fire on the button click, only once the fetch request has completed (under 10,000 transactions).

Perhaps `DOWNLOAD_CSV_CLICK` is misleading?

Reply:
Good spotting! The event should be fired as close to the user action as possible, i.e. `download_click`, in `onClick`. 

If we want to track actual successful downloads that's a different event (IMO we should just track the intent to download).

Reply:
> If we want to track actual successful downloads that's a different event

As this is an event that is already in place, maybe that was the intention and the renaming of the event/constant is the error.